sox.features.js

```diff
@@ -468,27 +468,10 @@
     shareLinksPrivacy: function() {
       // Description: Remove your user ID from the 'share' link
 
-      const questionTarget = document.getElementById('question');
-      const answersTargets = document.getElementsByClassName('answer');
-      sox.helpers.observe([questionTarget, ...answersTargets], '.share-tip', () => {
-        const toRemove = ' (includes your user id)';
-        const popup = document.getElementsByClassName('share-tip')[0];
-        if (!popup) return;
-
-        // Do nothing if the function's already done its thing
-        const popupTextInfo = popup.querySelector('.js-text');
-        const origText = popupTextInfo.innerText;
-        if (origText.indexOf(toRemove) == -1) return;
-
-        const input = popup.querySelector('input');
-        const origLink = input.value;
-        const newLink = origLink.match(/.+\/(q|a)\/[0-9]+/g)[0];
-
-        popupTextInfo.innerText = origText.replace(toRemove, '');
-
-        input.value = newLink;
-        input.select();
-      });
+      $('.js-subtitle').remove(); // Remove the 'includes your user id' string
+      for (let i = 0; i < $('.js-share-link').length; i++) {
+          $('.js-share-link:eq(' + i + ')').attr("href", $('.js-share-link:eq(' + i + ')').attr('href').match(/\/(q|a)\/[0-9]+/)[0]);
```

Comment:
Could you store the element returned by `$('.js-share-link:eq(' + i + ')')` in a separate variable at the top of the loop rather than selecting it twice? It's a tiny bit more readable and should be more efficient as we don't query the DOM twice!

Also if we store the length as another variable before the loop then the length lookup doesn't need to happen on every iteration, so it could be a bit more efficient.

Reply:
Actually looking at this again, could we use `$.each` here instead? I'm thinking this maybe:

```
$('.js-share-link').each((i, el) => {
    $(el).attr('href', $(el).attr('href').match(/\/(q|a)\/[0-9]+/)[0]);
});
```

or even without the jQuery!

Reply:
Oh, I didn't think about it! I'll try to work on that.